docs/DeveloperGuide.md

```diff
@@ -1,38 +1,253 @@
+
+Category: Category that stores the type tag of entry
+
+**Methods**
+
+Getters can be used to provide the private attributes to other classes
+
+editDescription(), editAmount(), editTime(), editCategory()
+
+* Takes in corresponding parameters to edit the private attributes.
+* Used by EntryList to make edits
+
+isSameMonth()
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+* Takes in Integer year and Month month and returns if the entry is made in the date specified by parameters.
 
+### EntryList
 
-## Product scope
-### Target user profile
+The EntryList class represents a collection of Entry instances. It is an abstract class that serves as a superclass for
+ExpenseList and IncomeList classes, providing common functionalities to minimize repetitive code and easing code
+maintenance. It provides underlying methods for adding, removing, editing, and listing entries from the list, which can
+be expenses or incomes depending on the subclass.
 
-{Describe the target user profile}
+On top of the methods that provide basic functionality of expense/income lists, the EntryList class provides additional
+methods such as summing all entries or filtering the entries according to LocalDate provided to the method. Most methods
+of EntryList take in a List<Entry>, which is because it has to operate on a list of expenses or incomes passed by the
+IncomeList or ExpenseList subclasses.
 
-### Value proposition
+**Methods**
 
-{Describe the value proposition: what problem does it solve?}
+addEntry()
 
-## User Stories
+* Takes an Entry instance as a parameter and adds it to the class-level list
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+listEntry()
 
-## Non-Functional Requirements
+* Takes in a list of Entry and passes them to the UI to print to the user.
+* Makes use of toString() method of entries.
+
+deleteEntry()
+
+* Takes in the list of entry and index of the entry to be deleted.
+* The method acts as an underlying method for subclasses' deletion methods.
+
+editEntry()
+
+* Overloaded method. All methods take in the LinkedList of entries and the index of the entry that has to be edited.
+  Another parameter has to be provided, which can be double, LocalDateTime, or String.
+* additional parameter double edits the money amount of the entry in the list.
+* additional parameter LocalDateTime edits the time when the entry is made.
+* additional parameter String edits the description of the entry in the list.
+
+getTotalAmount()
+
+* Takes in a list of entries and returns the sum of all entry's monetary value.
+
+SortEntryByAmount()
+
+* Returns a sorted list of entry by amount of entry.
+
+SortEntryByDate()
```

Comment:
are these 2 methods meant to be written as `sortEntryByAmount` and `sortEntryByDate()` instead? To follow the camelCase convention for method naming